sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs

```diff
@@ -15,15 +16,26 @@ public class TableServiceClient
         private readonly TableInternalClient _tableOperations;
         private readonly OdataMetadataFormat _format = OdataMetadataFormat.ApplicationJsonOdataFullmetadata;
 
+        public TableServiceClient(Uri endpoint, TableClientOptions options = null)
+        : this(endpoint, default, options) { }
+
         public TableServiceClient(Uri endpoint, TablesSharedKeyCredential credential, TableClientOptions options = null)
         {
             Argument.AssertNotNull(endpoint, nameof(endpoint));
-            Argument.AssertNotNull(credential, nameof(credential));
```

Comment:
In this constructor `credential` has to be not null might be better to add a common method/constructor both of these constructors would call.